heartbeat/monitors/wrappers/summarizer/summarizer.go

```diff
@@ -22,28 +22,53 @@ import (
 	"sync"
 	"time"
 
-	"github.com/gofrs/uuid"
-
-	"github.com/elastic/beats/v7/heartbeat/eventext"
 	"github.com/elastic/beats/v7/heartbeat/monitors/jobs"
+	"github.com/elastic/beats/v7/heartbeat/monitors/logger"
 	"github.com/elastic/beats/v7/heartbeat/monitors/stdfields"
 	"github.com/elastic/beats/v7/heartbeat/monitors/wrappers/monitorstate"
 	"github.com/elastic/beats/v7/libbeat/beat"
-	"github.com/elastic/elastic-agent-libs/logp"
-	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+// Summarizer produces summary events (with summary.* and other asssociated fields).
+// It accumulates state as it processes the whole event field in order to produce
+// this summary.
 type Summarizer struct {
 	rootJob        jobs.Job
 	contsRemaining uint16
 	mtx            *sync.Mutex
-	jobSummary     *JobSummary
-	checkGroup     string
-	stateTracker   *monitorstate.Tracker
 	sf             stdfields.StdMonitorFields
+	mst            *monitorstate.Tracker
 	retryDelay     time.Duration
+	plugins        []SummarizerPlugin
+	startedAt      time.Time
+}
+
+// EachEventActions is a set of options using bitmasks to inform execution after the EachEvent callback
+type EachEventActions uint8
+
+// DropErrEvent if will remove the error from the job return.
+const DropErrEvent = 1
+
+// OnSummaryActions is a set of options using bitmasks to inform execution after the OnSummary callback
+type OnSummaryActions uint8
+
+// RetryOnSummary will retry the job once complete.
+const RetryOnSummary = 1
+
+// SummarizerPlugin encapsulates functionality for the Summarizer that's easily expressed
+// in one location. Prior to this code was strewn about a bit more and following it was
+// a bit trickier.
+type SummarizerPlugin interface {
+	// EachEvent is called on each event, and allows for the mutation of events
+	EachEvent(event *beat.Event, err error) EachEventActions
+	// If at least one plugin returns true a retry will be performed
```

Comment:
nit: is this comment in the wrong place?

Reply:
Ah, that used to be true, but was refactored into an action